### PR TITLE
Fix rhcos image caching in CI

### PR DESCRIPTION
**Pull request by stbenjam**
I had an almost successful CI run this morning: http://10.8.144.11:8080/job/dev-tools/584/console.  The final step of CI seems to be the problem. This is the error:

```
+ for FILE in '$FILESTOCACHE'
++ basename /opt/dev-scripts/ironic/html/images/-openstack.qcow2
+ '[' '!' -f /opt/data/filecache/-openstack.qcow2 ']'
++ basename /opt/dev-scripts/ironic/html/images/-openstack.qcow2
+ sudo cp /opt/dev-scripts/ironic/html/images/-openstack.qcow2 /opt/data/filecache/-openstack.qcow2
cp: cannot stat ‘/opt/dev-scripts/ironic/html/images/-openstack.qcow2’: No such file or directory
```

The problem is we set `FILESTOCACHE` at the beginning of CI before we actually know the image name - we get that from the installer in step 3.  At the start of CI, `$RHCOS_IMAGE_FILENAME_OPENSTACK` isn't set... :chicken: :egg:

This fetches the last known image from the cache, and saves back the current image once it's known at the end.

### CODE REVIEW

**Comment by stbenjam** [comment]
Maybe there's a better way to do this, open to ideas...

File: run_ci.sh [hunk 1]
```diff
@@ -45,7 +45,9 @@ mount | grep root-
 # The CI host has a "/" filesystem that reset for each job, the only partition
 # that persist is /opt (and /boot), we can use this to store data between jobs
 FILECACHEDIR=/opt/data/filecache
-FILESTOCACHE="/opt/dev-scripts/ironic/html/images/$RHCOS_IMAGE_FILENAME_OPENSTACK /opt/dev-scripts/ironic/html/images/ironic-python-agent.initramfs /opt/dev-scripts/ironic/html/images/ironic-python-agent.kernel"
+# Get the last openstack image
+LAST_OPENSTACK_IMAGE=$(basename $(ls -r $FILECACHEDIR/*-openstack.qcow2 | head -n1))
```
Comment (russellb) [inline run_ci.sh]: I think we will still get the same error if the cache is empty. Maybe make sure the image exists here before adding it to FILESTOCACHE?

Comment (stbenjam) [reply run_ci.sh]: Oh, good point!

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/586/

**Pull request merged**